pkg/controller/resolvedofferings/resolvedofferings.go

```diff
@@ -0,0 +1,65 @@
+package resolvedofferings
+
+import (
+	"github.com/acorn-io/baaah/pkg/router"
+	internalv1 "github.com/acorn-io/runtime/pkg/apis/internal.acorn.io/v1"
+	"github.com/acorn-io/runtime/pkg/condition"
+	"github.com/acorn-io/runtime/pkg/config"
+)
+
+// Calculate is a handler that sets the resolved offerings for an AppInstance to its status if
+// and only if its generation is different from its observedGeneration.
+//
+// This is necessary because querying for resolved offerings will result in all running
+// AppInstances using that default to redeploy when a default changes. By
+// calculating the resolved offerings only when the generation changes, we can ensure that
+// updated resolved offerings are only applied when an AppInstance is updated directly.
+func Calculate(req router.Request, resp router.Response) (err error) {
+	appInstance := req.Object.(*internalv1.AppInstance)
+	status := condition.Setter(appInstance, resp, internalv1.AppInstanceConditionResolvedOfferings)
+
+	defer func() {
+		if err == nil {
+			status.Success()
+		} else {
+			status.Error(err)
+			// Disable prune because we are short-circuiting the handlers and don't want objects getting cleaned up accidentally.
+			resp.DisablePrune()
+			// Don't return the error so the other conditions don't get the same information.
+			err = nil
+		}
+	}()
+
+	// resolveVolumeClasses should run everytime as the function itself will not overwrite any existing
+	// information. Effectively, this means that volume class info only gets set if it have not been set before.
```

Comment:
The fact that this happens outside the ObservedGeneration gate, even if it's idempotent, falsifies the "if and only if" wording from the godoc comment of `Calculate()`:

> sets the resolved offerings for an AppInstance to its status if and only if its generation is different from its observedGeneration.

Reply:
Oh thanks, I forgot to update that comment on Calculate. We have to always resolve volume information in case the volume sizes change. I will fix the comment.